### PR TITLE
feat: 5865 add data node v2 apis

### LINKED ISSUE #5685: Add data-node V2 APIs

### LINKED ISSUE #5764: [graphql] transfersConnection returns an internal error
# Problem encountered
The graphql query `transfersConnection` is returning an internal error. This should never happen.

# Observed behaviour
On testnet, I placed a transfer:
https://explorer.fairground.wtf/txs/0x96A7196E251089935CF1AB4AFDE77486774E512BF3DE1AA58E578FF02EDBE73F
```json
{
  "fromAccountType": "ACCOUNT_TYPE_GENERAL",
  "to": "927b56614927b3a1c93f1707705acd4b3dbd0f6cd42f05a4e455ffbd84bacc35",
  "toAccountType": "ACCOUNT_TYPE_GENERAL",
  "asset": "5cfa87844724df6069b94e4c8a6f03af21907d7bc251593d08e4251043ee9f7c",
  "amount": "1000000",
  "oneOff": {}
}
```

- From party had enough assets in their general account for that asset
- To party did not previously exist
- The transfer succeeded, so now To party has the correct amount of asset in their general account

However querying for the transfer returns an error:

```graphql
{
  transfersConnection(pubkey: "6eed3515524b3c6dda7a9eceaebba58e236a56faa8ef7b8129179c9d269f41f2", direction: ToOrFrom) {
    edges {
      node {
        id
        from
        to
        asset { id name }
        amount
        status
      }
    }
  }
}
```

returns

```json
{
  "errors": [
    {
      "message": "rpc error: code = Internal desc = Internal error",
      "path": [
        "transfersConnection"
      ]
    }
  ],
  "data": null
}
```

# Expected behavior
I see a list of transfers

# Steps to reproduce

## Manual
Craft a transfer, along the lines of:
```bash
vegawallet command send --wallet "test" --pubkey "6eed3515524b3c6dda7a9eceaebba58e236a56faa8ef7b8129179c9d269f41f2" --network fairground {"transfer":"fromAccountType":4,"toAccountType":4, "to":"927b56614927b3a1c93f1707705acd4b3dbd0f6cd42f05a4e455ffbd84bacc35","asset":"5cfa87844724df6069b94e4c8a6f03af21907d7bc251593d08e4251043ee9f7c","amount":"1000000","oneOff"}:{"deliverOn":0}}}
```

Then go to a Playground and do the above query

## Automation
Link to automation and explanation on how to run it to reproduce the problem/bug

# Definition of Done
>ℹ️ Not every issue will need every item checked, however, every item on this list should be properly considered and actioned to meet the [DoD](https://github.com/vegaprotocol/vega/blob/develop/DEFINITION_OF_DONE.md).

**Before Merging**
- [ ] Code refactored to meet SOLID and other code design principles
- [ ] Code is compilation error, warning, and hint free
- [ ] Carry out a basic happy path end-to-end check of the new code
- [ ] All APIs are documented so auto-generated documentation is created
- [ ] All bug recreation steps can be followed without presenting the original error/bug
- [ ] All Unit, Integration and BVT tests are passing
- [ ] Implementation is peer reviewed (coding standards, meeting acceptance criteria, code/design quality)
- [ ] Create [front end](https://github.com/vegaprotocol/token-frontend/issues) or [console](https://github.com/vegaprotocol/console/issues) tickets with feature labels (should be done when starting the work if dependencies known i.e. API changes)

**After Merging**
- [ ] Move development ticket to `Done` if there is **NO** requirement for new [system-tests](https://github.com/vegaprotocol/system-tests/issues)
- [ ] Resolve any issues with broken system-tests
- [ ] Create [documentation](https://github.com/vegaprotocol/documentation/issues) tickets with feature labels if functionality has changed, or is a new feature

### PR DESCRIPTION
**Pull request by guoguojin**
Closes #5685
Closes #5764

### CODE REVIEW

File: script/postgres-docker.sh [hunk 1]
```diff
@@ -0,0 +1,7 @@
+#!/bin/bash
+docker run --rm \
+           -e POSTGRES_USER=vega \
+           -e POSTGRES_PASSWORD=vega \
+           -e POSTGRES_DB=vega \
+           -p 5432:5432 \
+           timescale/timescaledb:2.6.0-pg14
```
Comment (jeremyletang) [inline script/postgres-docker.sh:7]: Is that the last version?

File: datanode/entities/entities.go [hunk 2]
```diff
@@ -15,7 +15,7 @@ package entities
 import "google.golang.org/protobuf/proto"
 
 type PagedEntity[T proto.Message] interface {
-	Market | Party | Trade | Order | MarginLevels | MarketData | Reward | Candle | Deposit | Withdrawal | Asset | OracleSpec | OracleData | Position | LiquidityProvision | Vote | Transfer | AccountBalance | Proposal | Delegation | Node | NetworkParameter | Checkpoint | StakeLinking
+	Market | Party | Trade | Order | MarginLevels | MarketData | Reward | Candle | Deposit | Withdrawal | Asset | OracleSpec | OracleData | Position | LiquidityProvision | Vote | Transfer | AccountBalance | Proposal | Delegation | Node | NetworkParameter | Checkpoint | StakeLinking | NodeSignature | KeyRotation | ERC20MultiSigSignerAddedEvent | ERC20MultiSigSignerRemovedEvent
```
Comment (edd) [inline datanode/entities/entities.go:18]: **question (non-blocking)**:  I can see entities in here that are not reflected in the GraphQL schema. Could you check to see that all of them are? Specifically `ERC20MultiSigSignerAddedEvent` jumped out.

Comment (guoguojin) [reply datanode/entities/entities.go:18]: There are a few functions that are exposed on the gRPC level that haven't made it through to the GraphQL side of things. So far in this migration, we have only migrated existing functionality of the GraphQL and gRPC functions.

We did have a discussion yesterday about this yesterday, it shouldn't take much work to plumb in the gRPC functions to the GraphQL API, but for now I've not added anything "new" to the GraphQL API, I've only deprecated functions that needed to be replaced with the newer cursor based paginated function.

I hope that makes sense.

Comment (edd) [reply datanode/entities/entities.go:18]: It makes sense not to add new functionality, but we have recently had a few bugs that come about due to fields in the GRPC not being added/mapped/reflected properly in the GraphQL schema, resulting in the graphql server throwing errors (https://github.com/vegaprotocol/data-node/issues/594 for a recent example)

I guess I'm just asking for a sanity check that if these new fields are in GRPC, and I was subscribed in GraphQL to `busEvents`, and one of these was emitted, that it wouldn't break the subscription by emitting a server side error?

File: script/postgres-docker.sh (hunk 1 above)
Comment (guoguojin) [reply script/postgres-docker.sh:7]: No, that needs to be updated, I copied it from the old Data-Node project, will update

File: datanode/entities/entities.go (hunk 2 above)
Comment (guoguojin) [reply datanode/entities/entities.go:18]: Absolutely, there are some other plumbing that's missing in the GraphQL resolvers which I'm currently working through. I'll make sure everything is mapped correctly.

**Review by pgs31 — DISMISSED** [review]
It's a bit of a beast so I haven't been through with a fine toothed comb but it looks pretty sensible to me. I've got a PR incoming that simplifies the pagination stuff a bit but I've left the existing code in so it won't break

**Review by pscott31 — APPROVED** [review]

**Pull request merged**